### PR TITLE
platformio: 5.0.4 -> 5.1.1

### PR DESCRIPTION
**Pull request by oxzi**
###### Motivation for this change
Update platformio to its latest release. However, to get this changes working, the following pull requests needed to be merged first:

- [x] #112701, python3Packages.starlette: 0.13.8 -> 0.14.2
- [x] #114012, pythonPackages.json-rpc: init at 1.13.0
- [x] #116727, pythonPackages.ajsonrpc: init at 1.1.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/arduino/platformio/core.nix [hunk 1]
```diff
@@ -1,9 +1,26 @@
 { stdenv, lib, buildPythonApplication, fetchpatch
-, bottle, click, colorama, semantic-version
-, lockfile, pyserial, requests
-, tabulate, pyelftools, marshmallow
-, pytest, tox, jsondiff
-, git, spdx-license-list-data
+
+, bottle
```
Comment (SuperSandro2000) [inline pkgs/development/arduino/platformio/core.nix:3]:
```suggestion
, bottle
```

File: pkgs/development/arduino/platformio/core.nix [hunk 2]
```diff
@@ -1,9 +1,26 @@
...
+, semantic-version
+, spdx-license-list-data
+, starlette
+, tabulate
+, tox
+, uvicorn
+, wsproto
+, zeroconf
+
 , version, src
```
Comment (SuperSandro2000) [inline pkgs/development/arduino/platformio/core.nix:24]:
```suggestion
, zeroconf
, version, src
```

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by oxzi** [comment]
I have updated and rebased this pull request for PlatformIO's latest release 5.1.1. This introduced a new dependency, ajsonrpc, for which I created another PR #116727. However, to make this changes work, the starlette package update needs to be merged first, #112701.

I have successfully tested this package with those two changes applied locally.

**Comment by oxzi** [comment]
After all necessary dependencies were merged, I rebased this PR once again and now it should be ready for a final review.

**Comment by leonardp** [comment]
FWIW I have also tested this locally and it is working as intended.

**Comment by oxzi** [comment]
@SuperSandro2000: Would you please take another look? Thanks!

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 114014` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Comment by happysalada** [comment]
The PR looks good to me, doesn't seem to be a major change anyway.
I'm going to go ahead and merge this considering everybody is busy with the coming release

**Pull request merged**